cli.py

```diff
@@ -219,7 +219,7 @@ def create(agent_name):
     import re
     import shutil
 
-    if not re.match("\W*$", agent_name):
+    if not re.match("\w*$", agent_name):
         click.echo(
             click.style(
                 f"😞 Agent name '{agent_name}' is not valid. It should not contain spaces or special characters other than -_",
```

Comment:
One thing to keep in mind when using the `\w` word meta-character is that it only matches alphanumeric characters and underscores. If you want to keep the original behaviour of also supporting hyphens(`-`) in the name, I would propose to simply revert the original PR introducing the meta-character: https://github.com/Significant-Gravitas/AutoGPT/pull/6096